wagtail/wagtailimages/views/serve.py

```diff
@@ -0,0 +1,68 @@
+from wsgiref.util import FileWrapper
+import imghdr
+import base64
+import hmac
+import hashlib
+
+from django.shortcuts import get_object_or_404
+from django.http import HttpResponse, StreamingHttpResponse, HttpResponsePermanentRedirect
+from django.core.exceptions import PermissionDenied, ImproperlyConfigured
+from django.views.generic import View
+from django.conf import settings
+from django.utils.six import text_type
+
+from wagtail.wagtailimages.models import get_image_model, SourceImageIOError
+from wagtail.wagtailimages.exceptions import InvalidFilterSpecError
+
+
+def generate_signature(image_id, filter_spec, key=None):
+    if key is None:
+        key = settings.SECRET_KEY
+
+    # Key must be a bytes object
+    if isinstance(key, text_type):
+        key = key.encode()
+
+    # Based on libthumbor hmac generation
+    # https://github.com/thumbor/libthumbor/blob/b19dc58cf84787e08c8e397ab322e86268bb4345/libthumbor/crypto.py#L50
+    url = str(image_id) + '/' + str(filter_spec) + '/'
+    return base64.urlsafe_b64encode(hmac.new(key, url.encode(), hashlib.sha1).digest())
+
+
+def verify_signature(signature, image_id, filter_spec, key=None):
+    return signature == generate_signature(image_id, filter_spec, key=key)
+
+
+class ServeView(View):
+    model = get_image_model()
+    action = 'serve'
+    key = None
+
+    def get(self, request, signature, image_id, filter_spec):
+        if not verify_signature(signature.encode(), image_id, filter_spec, key=self.key):
+            raise PermissionDenied
+
+        image = get_object_or_404(self.model, id=image_id)
+
+        # Get/generate the rendition
+        try:
+            rendition = image.get_rendition(filter_spec)
+        except SourceImageIOError:
+            return HttpResponse("Source image file not found", content_type='text/plain', status=410)
+        except InvalidFilterSpecError:
+            return HttpResponse("Invalid filter spec: " + filter_spec, content_type='text/plain', status=400)
```

Comment:
This is incorrect. Simple concatenation is much more performant than `.format()`:

```python
>>> timeit.timeit('"foo" + bar', setup="bar = 'bar'")
0.0954943690012442
>>> timeit.timeit('"foo{}".format(bar)', setup="bar = 'bar'")
0.25878583200028515
>>> timeit.timeit('"/{}/{}".format(bar, bar)', setup="bar = 'bar'")
0.2846537450022879
```

`.format()` is easier to read, and is more convenient than concatenation for complex strings, but it is not more performant. Simple concatenation like in the above code is fine as it is in my opinion.